server/src/domain/activity/activity.service.ts

```diff
@@ -0,0 +1,91 @@
+import { Inject, Injectable, Logger } from '@nestjs/common';
+import { AccessCore, Permission } from '../access';
+import { AuthUserDto } from '../auth';
+import { IAccessRepository, IActivityRepository } from '../repositories';
+import {
+  ActivityCommentDto,
+  ActivityDto,
+  ActivityLikeDto,
+  ActivityLikeStatusResponseDto,
+  ActivityResponseDto,
+  ActivityStatisticsResponseDto,
+  mapActivity,
+} from './activity.dto';
+
+@Injectable()
+export class ActivityService {
+  private access: AccessCore;
+  private logger = new Logger(ActivityService.name);
+
+  constructor(
+    @Inject(IAccessRepository) accessRepository: IAccessRepository,
+    @Inject(IActivityRepository) private repository: IActivityRepository,
+  ) {
+    this.access = AccessCore.create(accessRepository);
+  }
+
+  async getAll(authUser: AuthUserDto, dto: ActivityDto): Promise<ActivityResponseDto[]> {
+    await this.access.requirePermission(authUser, Permission.ALBUM_READ, dto.albumId);
+    if (dto.assetId) {
+      await this.access.requirePermission(authUser, Permission.ASSET_READ, dto.assetId);
+    }
+    const activities = await this.repository.search({ albumId: dto.albumId, assetId: dto.assetId });
+    return activities.map(mapActivity);
+  }
+
+  async getStatistics(authUser: AuthUserDto, dto: ActivityDto): Promise<ActivityStatisticsResponseDto> {
+    await this.access.requirePermission(authUser, Permission.ALBUM_READ, dto.albumId);
+    return { comments: await this.repository.getStatistics(dto.assetId, dto.albumId) };
+  }
+
+  async getLikeStatus(authUser: AuthUserDto, dto: ActivityDto): Promise<ActivityLikeStatusResponseDto> {
+    await this.access.requirePermission(authUser, Permission.ACTIVITY_CREATE, dto.albumId);
+    const [reaction] = await this.repository.search({
+      albumId: dto.albumId,
+      userId: authUser.id,
+      assetId: dto.assetId,
+    });
+    return { value: !!reaction };
+  }
+
+  async updateLikeStatus(authUser: AuthUserDto, dto: ActivityLikeDto): Promise<ActivityLikeStatusResponseDto> {
+    await this.access.requirePermission(authUser, Permission.ACTIVITY_CREATE, dto.albumId);
+
+    const options = {
+      userId: authUser.id,
+      albumId: dto.albumId,
+      assetId: dto.assetId,
+    };
+
+    const [reaction] = await this.repository.search(options);
```

Comment:
I think this needs to be reactions, and then reaction is find of one that isn't a comment. Or, include isLiked in the query. Missed this cleaning up earlier.